File: server/src/controllers/me/me.controller.ts
import { Request, Response, NextFunction } from "express";
import { isAdminLogin } from "../../config.js";

export default class MeController {
  getMe = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
    try {
      const user = req.session?.user ?? null;
      const isAdmin = isAdminLogin(user?.login);
      res.json({ user, isAdmin });
    } catch (error) {
      next(error);
    }
  };
}

File: server/Sources/App/Controllers/MeController.swift
import Vapor

struct MeController: RouteCollection {
    func boot(routes: RoutesBuilder) throws {
        routes.get(use: getMe)
    }

    func getMe(req: Request) async throws -> MeResponse {
        let user = req.sessionUser
        return MeResponse(user: user, isAdmin: AppConfig.isAdminLogin(user?.login))
    }
}
